psa-crypto/src/operations/mac.rs

```diff
@@ -2,3 +2,134 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! # Message Authentication Code (MAC) operations
+
+use crate::initialized;
+use crate::types::key::Id;
+use crate::types::algorithm::Mac;
+use crate::types::status::{Result, Status, Error};
+
+
+/// Calculate the message authentication code (MAC) of a message
+/// The key must allow `sign_message`
+///
+/// # Example
+///
+/// ```
+/// # use psa_crypto::operations::{mac::compute_mac, key_management::generate};
+/// # use psa_crypto::types::algorithm::{Hash, Mac, FullLengthMac};
+/// # use psa_crypto::types::key::{Attributes, Type, Lifetime, Policy, UsageFlags};
+/// # const MESSAGE: [u8; 32] = [
+/// #     0x69, 0x3E, 0xDB, 0x1B, 0x22, 0x79, 0x03, 0xF4, 0xC0, 0xBF, 0xD6, 0x91, 0x76, 0x37, 0x84, 0xA2,
+/// #     0x94, 0x8E, 0x92, 0x50, 0x35, 0xC2, 0x8C, 0x5C, 0x3C, 0xCA, 0xFE, 0x18, 0xE8, 0x81, 0x37, 0x78,
+/// # ];
+/// # let mut attributes = Attributes {
+/// #     key_type: Type::RsaKeyPair,
+/// #     bits: 1024,
+/// #     lifetime: Lifetime::Volatile,
+/// #     policy: Policy {
+/// #         usage_flags: UsageFlags {
+/// #             sign_message: true,
+/// #             ..Default::default()
+/// #         },
+/// #         permitted_algorithms: Mac::FullLength(FullLengthMac::Hmac{hash_alg: Hash::Sha256}).into(),
```

Comment:
Would be nice to have a `From<FullLength>` implementation on `FullLengthMac` to directly do:
```rust
FullLengthMac::Hmac{hash_alg: Hash::Sha256}.into()
```